### PR TITLE
Feature selection UI refactor

### PR DESCRIPTION
**Pull request by juandjara**
# Description

Shortcut: https://app.shortcut.com/cartoteam/story/315828/refactor-feature-selection-to-extract-ui-components-in-c4r [sc-315828]

Refactor FeatureSelectionWidgetUI split into many exported pieces + design revamp

## Type of change

(choose one and remove the others)

- Refactor

# Acceptance

This PR is needed to close project for drawing geometries in Workflows GeoJSON Editor.
This feature deals with theme / UI or internal elements used also in CARTO 3

More details to de defined...

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

**Review by Josmorsot — APPROVED** [review]
LGTM

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIToggleButton.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    </ClickAwayListener>
+  );
+}
+
+FeatureSelectionUIToggleButton.propTypes = {
+  className: PropTypes.string,
+  sx: PropTypes.any,
+  icon: PropTypes.element.isRequired,
+  hoverTooltip: PropTypes.string.isRequired,
+  clickTooltip: PropTypes.string.isRequired,
```
Comment (vmilan) [inline packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIToggleButton.js:88]: Maybe a more explanatory names?

```
tooltipText
enabledTooltipText
```

¿?

File: packages/react-ui/src/types.d.ts [hunk 2]
```diff
@@ -158,20 +157,54 @@ export type NoDataAlert = {
...
+  editModes?: FeatureSelectionWidgetUIData[];
+  selectedMode: string;
+  onSelectMode?: Function;
+  enabled?: boolean;
+  onEnabledChange?: Function;
+  geometry?: GeoJSON.Feature;
+  onSelectGeometry?: Function;
+  onDeleteGeometry?: Function;
+  className?: string;
+  sx?: SxProps<Theme>;
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply packages/react-ui/src/types.d.ts:173]: Agree about className, not sure about sx

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        <Box ml={2}>
+          <Typography variant='body2'>{capitalize(mode.label)}</Typography>
+        </Box>
+      </Box>
+    </MenuItem>
+  );
+
+  return (
+    <Box sx={sx} className={className}>
+      <Tooltip title='Choose a tool' placement={tooltipPlacement}>
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js:89]: Not sure, is it going to be used differently on both clients? current tool and the new from Builder?

**Review by VictorVelarde — COMMENTED** [review]
Overall LGTM, my only doubt is about sx injection of props

**Review by vmilan — APPROVED** [review]
LGTM, just added some minor comments. Thanks!!

File: packages/react-ui/src/types.d.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (juandjara) [reply packages/react-ui/src/types.d.ts]: props removed

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      <Menu
+        id='feature-selection-menu'
+        style={{ zIndex: theme.zIndex.tooltip + 1 }}
+        anchorEl={anchorEl}
+        open={open}
+        onClose={closeDropdown}
+        MenuListProps={{ 'aria-labelledby': 'feature-selection-menu-button' }}
+      >
+        <MenuItem disabled>
+          <Typography variant='caption'>Choose a drawing tool</Typography>
```
[the remark this answers is not in the document]
Comment (juandjara) [reply packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js]: copy text extracted

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionWidgetUI.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          disabledChipTooltip='Clear mask'
+          tooltipPlacement={tooltipPlacement}
+        />
+      )}
+    </StylesWrapper>
+  );
+}
+
+FeatureSelectionWidgetUI.defaultProps = {
+  sx: undefined,
```
[the remark this answers is not in the document]
Comment (juandjara) [reply packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionWidgetUI.js]: props removed

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        id='feature-selection-menu'
+        style={{ zIndex: theme.zIndex.tooltip + 1 }}
+        anchorEl={anchorEl}
+        open={open}
+        onClose={closeDropdown}
+        MenuListProps={{ 'aria-labelledby': 'feature-selection-menu-button' }}
+      >
+        <MenuItem disabled>
+          <Typography variant='caption'>{menuHeaderText}</Typography>
+        </MenuItem>
```
[the remark this answers is not in the document]
Comment (juandjara) [reply packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js:114]: Ok, updated this too

File: packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (juandjara) [reply packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionUIDropdown.js]: Text extracted

**Pull request merged**